src/gs-shell-overview.c

```diff
@@ -194,9 +206,15 @@ gs_shell_overview_get_popular_cb (GObject *source_object,
 	}
 	gs_app_list_randomize (list);
 
-	/* Sort the randomized list based on the presence of the popular
+	sort_alphabetically = g_settings_get_boolean (priv->settings,
+						      "sort-overview-alphabetically");
+	/* In any case always sort the randomized list based on the presence of the popular
 	 * background so it gives precedence to apps that have that asset */
-	gs_app_list_sort (list, sort_with_popular_background, NULL);
+	if (!sort_alphabetically) {
```

Comment:
Normally, I would probably invert the polarity of the check to check for `if (sort_alphabetically)`, but not a big deal, so I'm going to merge as is.